# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../uclibc/patches/vdr-realpath.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

man canonicalize_file_name:

DESCRIPTION
       The call canonicalize_file_name(path) is equivalent to the call realpath(path, NULL).

CONFORMING TO
       The function is a GNU extension.

Calling realpath with a NULL target pointer is likewise a
non-standard glibc extension which is not supported by uClibc.

  -- mtr <michael@tross.org>

--- vdr-1.4.3/tools.c.orig	2006-08-12 15:30:07.000000000 +0200
+++ vdr-1.4.3/tools.c	2006-11-03 11:44:12.000000000 +0100
@@ -481,7 +481,8 @@
 {
   if (!FileName)
      return NULL;
-  char *TargetName = canonicalize_file_name(FileName);
+  char *TargetName = MALLOC(char, PATH_MAX);
+  TargetName = realpath(FileName, TargetName);
   if (!TargetName) {
      if (errno == ENOENT) // file doesn't exist
         TargetName = strdup(FileName);
